### ISSUE TITLE
Unexpected behaviour when test name contains equals = character

### ISSUE THREAD

**Issue by Sjord** [tree-sitter/tree-sitter#1047]
Having `=` in the title of a test makes the test before it fail.

## Reproduction

In [tree-sitter-php/test/corpus/class.txt](https://github.com/tree-sitter/tree-sitter-php/blob/master/test/corpus/class.txt#L38), change `Anonymous classes` to `Anonymous=classes` and run `tree-sitter test`.

## Actual behaviour

First, no test `Anonymous=classes` shows: 

```
  class:
    ✗ Abstract class
    ✓ Conditional class definition
    ✓ Class constant modifiers
    ✓ Final class
    ✓ Implicitly public properties and methods
    ✓ Property Types
```

Second, the test before that fails in a peculiar way:

```
  1. Abstract class:
    (program (php_tag) (class_declaration (class_modifier) (name) (declaration_list (method_declaration (visibility_modifier) (name) (formal_parameters) (compound_statement)) (method_declaration (class_modifier) (visibility_modifier) (name) (formal_parameters)))) (ERROR) (expression_statement (binary_expression (binary_expression (unary_op_expression (parenthesized_expression (function_call_expression (function_call_expression (qualified_name (name)) (arguments (qualified_name (name)))) (arguments (function_call_expression (function_call_expression (function_call_expression (qualified_name (name)) (arguments (qualified_name (name)))) (arguments (qualified_name (name)))) (arguments (function_call_expression (function_call_expression (qualified_name (name)) (arguments (function_call_expression (function_call_expression (function_call_expression (function_call_expression (qualified_name (name)) (arguments (qualified_name (name)))) (arguments (qualified_name (name)))) (arguments (qualified_name (name)))) (arguments (qualified_name (name)))))) (arguments (function_call_expression (function_call_expression (function_call_expression (function_call_expression (qualified_name (name)) (arguments (qualified_name (name)))) (arguments (qualified_name (name)))) (arguments (qualified_name (name)))) (arguments (qualified_name (name)))))))))))) (ERROR) (qualified_name (name))) (ERROR (name)) (ERROR (php_tag)) (object_creation_expression (declaration_list (method_declaration (visibility_modifier) (name) (formal_parameters) (compound_statement)))))) (compound_statement))))) (expression_statement (object_creation_expression (base_clause (qualified_name (name))) (class_interface_clause (qualified_name (name)) (qualified_name (name))) (declaration_list))) (expression_statement (object_creation_expression (arguments) (declaration_list (property_declaration (visibility_modifier) (property_element (variable_name (name))))))) (expression_statement (object_creation_expression (arguments (variable_name (name)) (variable_name (name))) (base_clause (qualified_name (name))) (declaration_list (use_declaration (qualified_name (name)))))) (class_declaration (name) (declaration_list (method_declaration (visibility_modifier) (name) (formal_parameters) (compound_statement (return_statement (object_creation_expression (arguments (variable_name (name))) (base_clause (qualified_name (name))) (declaration_list (const_declaration (const_element (name) (string))))))))))) 
```

## Expected behaviour

I expect `Anonymous=classes` to be a valid name for a test, that it runs like any other test, and I don't expect it to have an effect on other tests.

**Comment by Sjord** [comment]
[test.rs](https://github.com/tree-sitter/tree-sitter/blob/master/cli/src/test.rs#L19) identifies headers with the regex `^===+\r?\n([^=]*)\r?\n===+\r?\n`, which explicitly disallows `=` in titles.

**Comment by ahelwer** [comment]
This could be made possible with implementation of #982 since the addition of suffixes would enable using a regex to match arbitrary non-newline characters for the test name.

**Comment by Sjord** [comment]
Solved by PR #1348.
